agent/node.go

```diff
@@ -622,15 +622,18 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 
 			select {
 			case <-ctx.Done():
+				m.Stop(context.Background()) // todo: this should be sync like other components
```

Comment:
Note: this line is being moved because if we remove a node from raft, we want its raft state machine to see that and move its state directories out of the way to prepare for later promotion. If we stop the manager when we notice the change to the spec, this won't work properly.